community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java

```diff
@@ -0,0 +1,385 @@
+        boolean updateConf = arguments.getBoolean( ARG_UPDATE_CONF );
+        boolean overwrite = arguments.getBoolean( ARG_OVERWRITE );
+        String heap = bytesToString( recommendHeapMemory( memory ) );
+        String pagecache = bytesToString( recommendPageCacheMemory( memory ) );
+
+        if ( printConf )
+        {
+            outputUpdatedConf( confFile, heap, pagecache, overwrite, outsideWorld::stdOutLine );
+        }
+        else
+        {
+            printRecommendation( memoryArgument, heap, pagecache );
+        }
+
+        if ( updateConf )
+        {
+            ArrayList<String> updatedLines = new ArrayList<>();
+            outputUpdatedConf( confFile, heap, pagecache, overwrite, updatedLines::add );
+            try
+            {
+                Files.write( confFile, updatedLines, StandardCharsets.UTF_8 );
+                outsideWorld.stdErrLine( "File updated: " + confFile );
+            }
+            catch ( IOException e )
+            {
+                throw new CommandFailed( "Failed to update configuration file: " + confFile, e );
+            }
+        }
+    }
+
+    private void outputUpdatedConf(
+            Path confFile, String heap, String pagecache, boolean overwrite, Consumer<String> out ) throws CommandFailed
+    {
+        try
+        {
+            // Existing settings to maybe skip.
+            Map<String,String> existingSettings = MapUtil.load( confFile.toFile() );
+            // Settings that were updated by our pass-through.
+            Set<String> redefined = new HashSet<>();
+            // Skip settings that are either already defined in the conf file, or were defined earlier in our pass.
+            Predicate<String> skip = s -> existingSettings.containsKey( s ) || redefined.contains( s );
+            try ( Stream<String> lines = Files.lines( confFile, StandardCharsets.UTF_8 ) )
+            {
+                lines.forEachOrdered( line ->
+                {
+                    line = tryUpdateSetting( line, skip, redefined, overwrite, initialHeapSize.name(), heap );
+                    line = tryUpdateSetting( line, skip, redefined, overwrite, maxHeapSize.name(), heap );
+                    line = tryUpdateSetting( line, skip, redefined, overwrite, pagecache_memory.name(), pagecache );
+                    out.accept( line );
+                } );
+            }
+            defineIfMissing( out, redefined, initialHeapSize.name(), heap );
+            defineIfMissing( out, redefined, maxHeapSize.name(), heap );
+            defineIfMissing( out, redefined, pagecache_memory.name(), pagecache );
+        }
+        catch ( Exception e ) // Also catch runtime exceptions due to use of Streams.
+        {
+            throw new CommandFailed( "Failed to read configuration file: " + confFile, e );
+        }
+    }
+
+    private String tryUpdateSetting( String line, Predicate<String> alreadyDefined, Set<String> redefined,
```

Comment:
this method and related trying to do a bunch of work by parsing and trying to understand user config file, in the same time it will fail to do so in some of the cases that we support during normal config loading since the method does not cover all possible format possibilities.
For example, multi-line lines or `:` as a separator just to name a few, so maybe instead of doing this attempt we will perform additional validation of already configured values or generate a suggested property values, so users can update config file if they think its relevant for them.
What do you guys think?

Reply:
Personally I'm all for it. I think it would be sensible to do 

```
sudo neo4j-admin memrec >> /etc/neo4j/neo4j.conf
```

but I know support will really like an in-place update feature.

Reply:
Taking your example literally, I think we can all find a solution that can satisfy us all:

```
sudo neo4j-admin memrec >> /etc/neo4j/neo4j.conf
```

All this needs to do, really, is just produce a _section_ of configuration file text, that can be appended to the end of an existing (or not) configuration file. There is no need parse any existing file. We just have to produce output in a format that works in a configuration file.

Reply:
my original example was incorrect. I naturally meant

    neo4j-admin memrec | sudo tee -a /etc/neo4j/neo4j.conf

But yes. The way the config *should* work is that the last defined value is used in case of duplicate values, so there should not be any need to change anything in the file.

That is of course how it *should* be. I know Java does that, I am less certain about our wrapper scripts parsing of the config. That's a separate issue though.